### DISCUSSION TITLE
Best way to tie into SQLAlchemy to send events on model changes?

### DISCUSSION THREAD

**Discussion by jamesrusso** [sqlalchemy/sqlalchemy#11396] [Usage Questions]
I am working on an application where I would like to notify the web clients of updates. This is done via web sockets using a service such as pusher. Right now, I've wired things into our services, but I am thinking that I could likely better integrate this into SQLAlchemy using events?  These events need to only fire after the commit. 

I am thinking that I would attach to the after_update and after_insert, I would then store the events in a dictionary associated with the session and then link the after commit and fire those events? 

Has anyone done anything similar? Bad idea?

**Comment by rhildred** [comment]
Hi:

Thanks for this. Did you get anywhere with your idea? I am also interested in the same thing. Interested in sqlalchemy because of alembic. Feel that with this feature sql alchemy woukd be a devops firebase/supabase.

$.02

Rich

**Reply by CaselIT** [reply]
Hi, I think sqlalchemy has all the required events that you can register to track the required interaction to build such notification.

It will necessarily need to leverage 3rd party libraries to notify http clients and whatnot since those are all outside the scope of sqlalchemy.

You can refer to the event documentation: https://docs.sqlalchemy.org/en/20/orm/events.html#orm-events
